apps/survey/app/src/math-utils.test.js

```diff
@@ -0,0 +1,55 @@
+import BigNumber from 'bignumber.js'
+import { scaleBigNumberValuesSet } from './math-utils'
+
+const scaleResults = (values, total) =>
+  scaleBigNumberValuesSet(values, new BigNumber(total)).map(v => v.toFixed(0))
+
+describe('scaleValuesSet()', () => {
+  const sets = [
+    [
+      'should add round up the first number first',
+      [[0.5, 0.5], '329879'],
+      ['164940', '164939'],
+    ],
+    [
+      'should work with very big numbers',
+      [[0.5, 0.5], '3298792983798273972398792837972310987189327'],
+      [
+        '1649396491899136986199396418986155493594664',
+        '1649396491899136986199396418986155493594663',
+      ],
+    ],
+    [
+      'should return an empty array if an empty array is provided',
+      [[], '100'],
+      [],
+    ],
+    [
+      'should handle zero values',
+      [[0.9, 0, 0.1], '10000'],
+      ['9000', '0', '1000'],
+    ],
+    [
+      'should correct the values if their sum is less than 1',
+      [[0.1, 0.1], '1000'],
+      ['900', '100'],
+    ],
+    [
+      'should correct the values if their sum is more than 1',
+      [[1.1, 0.3], '1000'],
+      ['700', '300'],
```

Comment:
These two behaviours seem like they'd be surprising—given `[0.1, 0.1]`, I would expect `['100', '100']`, and given `[1.1, 0.3]` I would expect an error similar to the `'too far from 1'` error.

Reply:
This is to correct any number precision issue when summing the values to get `1`, but I agree it is not ideal.

Do you think it would be a good idea to check if `Math.abs(1 - sum) < 0.0001`?

- A value within that range would be accepted as `1`.
- A value below would give a result below 100% (`[0.1, 0.1]` => `['100', '100']`).
- A value above would trigger a `too far from 1` error.

Reply:
@bpierre I think those rules make a lot of sense, especially if we let the user configure the precision they need.